src/leaflet.filelayer.js

```diff
@@ -303,7 +321,7 @@
             fileInput.style.display = 'none';
             // Load on file change
             fileInput.addEventListener('change', function () {
-                thisFileLayerLoad._loadFiles(this.files);
+                thisLoader.loadMultiple(this.files);
```

Comment:
This was in the original FileLayerLoad._loadFiles() method, I just moved it.
But you point is good, and I actually tested without the timeout, it works fine. Code is now cleaner (no more extra 'reader' passed to load() for tests).
I just hope this timeout was not introduced for some good reason... All tests are passing though.